diff --git a/lib/healpy/sphtfunc.py b/lib/healpy/sphtfunc.py
--- a/lib/healpy/sphtfunc.py
+++ b/lib/healpy/sphtfunc.py
@@ -1201,7 +1201,10 @@ def harmonic_ud_grade(
             beam_window_out = beam_window_out[: lmax + 1, :]
 
     if use_pixel_weights and input_type == "map":
-        filename = "full_weights/healpix_full_weights_nside_%04d.fits" % nside_in
+        filename = os.path.join(
+            "full_weights",
+            "healpix_full_weights_nside_%04d.fits" % nside_in,
+        )
         if datapath is not None:
             pixel_weights_filename = os.path.join(datapath, filename)
             if not os.path.exists(pixel_weights_filename):
